ddtrace/sampler.py

```diff
@@ -205,7 +207,7 @@ def __init__(
         if default_sample_rate is None:
             log.debug("initialized DatadogSampler, limit %r traces per second", rate_limit)
             # Default to previous default behavior of RateByServiceSampler
-            self.default_sampler = RateByServiceSampler()  # type: BaseSampler
+            self.default_sampler = RateByServiceSampler()  # type: Union[RateByServiceSampler, SamplingRule]
```

Comment:
This is probably how it was before but just wanted to call out how strange it is for a "sampler" to possibly be a "SamplingRule"

Reply:
They both implement `BaseSampler`, but I wanted to be more restrictive since it literally can only be a `RateByServiceSampler` or `SamplingRule`

Maybe `RuleSampler` is a better name for it? but it *is* a `BaseSampler`